### DISCUSSION TITLE
Remote Playback API

### DISCUSSION THREAD

**Discussion by luwes** [muxinc/media-chrome#654] [Proposals]
Media Chrome could make use of the https://developer.mozilla.org/en-US/docs/Web/API/Remote_Playback_API (API interface)
for casting and airplay which we discovered after writing https://github.com/muxinc/castable-video (thanks @bdougherty)

https://beaufortfrancois.github.io/sandbox/media/remote-playback.html
https://avayvod.github.io/remote-playback/test.html

It seems to be implemented for both iOS and Android but no desktop.

Should be pretty easy to patch the API on desktop so it uses the Airplay API internally for Safari desktop and hook up the castable-video API which will be a more standardized API than the one we came up with.

---

The Vimeo player.js API has implemented a similarly shaped API 
https://github.com/vimeo/player.js/commit/aebeed5dd7255cd2f05043539fa18909f6edfe62

**Comment by bdougherty** [comment]
It does actually work in Safari on Mac for AirPlay. It's Chromium that hasn't implemented at all on desktop for Chromecast, even though they claim it is shipping (I guess just having the function even though it is a no-op counts for them).

**Comment by luwes** [comment]
Released in v2 https://github.com/muxinc/media-chrome/releases/tag/v2.0.0 🎉
